Source/player.cpp

```diff
@@ -1055,10 +1052,10 @@ bool DoAttack(Player &player)
 			}
 		}
 
-		if (dMonster[dx][dy] != 0) {
-			didhit = PlrHitMonst(player, abs(dMonster[dx][dy]) - 1);
-		} else if (dPlayer[dx][dy] != 0 && !player.friendlyMode) {
-			didhit = PlrHitPlr(player, Players[abs(dPlayer[dx][dy]) - 1]);
+		if (monster != nullptr) {
+			didhit = PlrHitMonst(player, *monster);
+		} else if (monster != nullptr && !player.friendlyMode) {

```

Comment:
monster  != nullptr is the wrong check here, with that change this case will never be true.
```suggestion
		} else if (IsPlayerAtPosition(position) && !player.friendlyMode) {
```

Reply:
Thanks!

Reply:
You will also have to implement `IsPlayerAtPosition`

Reply:
I replaced it with `PlayerAtPosition`